extensions/wikia/AuthModal/js/AuthModal.js

```diff
@@ -3,27 +3,57 @@ define('AuthModal', ['jquery', 'AuthComponent', 'wikia.window'], function ($, Au
 
 	var modal,
 		$blackout,
-		language = window.wgContentLanguage;
+		isOpen,
+		language = window.wgContentLanguage,
+		track;
 
 	function open () {
+		if (isOpen) {
+			close();
+		}
 		$('.WikiaSiteWrapper').append(
-			'<div class="auth-blackout blackout visible"><div class="auth-modal loading"><a class="close"></div></div>'
+			'<div class="auth-blackout blackout visible"><div class="auth-modal loading">' +
+				'<a class="close" href="#"></div></div>'
 		);
+		isOpen = true;
 		$blackout = $('.auth-blackout');
 		modal = $blackout.find('.auth-modal')[0];
 		$('.auth-blackout, .auth-modal .close').click(close);
+
+		track = getTrackingFunction(Wikia.Tracker.ACTIONS.OPEN);
```

Comment:
Do we need to build tracking function every time we open modal?
```action``` is not shared between different calls, what do you think about moving it to ```track``` calls?